PyInstaller/compat.py

```diff
@@ -24,6 +24,23 @@
 import sys
 import shutil
 
+# PyInstaller requires importlib.metadata from python >= 3.10 stdlib, or equivalent importlib_metadata >= 4.6.
+if sys.version_info >= (3, 10):
+    import importlib.metadata as importlib_metadata
+else:
+    try:
+        import importlib_metadata
+    except ModuleNotFoundError as e:
+        from PyInstaller.exceptions import ImportlibMetadataError
+        raise ImportlibMetadataError() from e
+
+    import packaging.version  # For importlib_metadata version check
+
+    # Validate the version
+    if packaging.version.parse(importlib_metadata.version("importlib-metadata")) < packaging.version.parse("4.6"):
```

Comment:
True, but that is only at PyInstaller's installation time. Nothing really prevents a determined user (or possibly a misconfigured package) from installing a different version of `importlib-metadata` later on. And I would prefer to shut down such nonsense as soon as possible instead of dealing with tracebacks about `importlib_metadata` missing `packages_distributions` or `importlib_metadata.entry_points` not having a `groups` argument.